src/middlewared/middlewared/plugins/sysdataset.py

```diff
@@ -557,18 +511,22 @@ async def migrate(self, _from, _to):
         await self.__nfsv4link(config)
 
 
+async def pool_post_create(middleware, pool):
+    if (await middleware.call('systemdataset.config'))['pool'] == await middleware.call('boot.pool_name'):
+        await middleware.call('systemdataset.setup')
+
+
 async def pool_post_import(middleware, pool):
     """
     On pool import we may need to reconfigure system dataset.
     """
-    if pool is None:
-        return
-
     await middleware.call('systemdataset.setup')
 
 
 async def setup(middleware):
-    middleware.register_hook('pool.post_import', pool_post_import, sync=True)
+    middleware.register_hook('pool.post_create', pool_post_create)
```

Comment:
Do we now need to look (or add) a `pool.post_{delete/remove/export}` hook for when a pool is exported/destroyed etc and the system dataset exists on that pool? I think we already have one but wondering if it needs to be updated with these changes?

Reply:
@yocalebo great idea, done